fenix/app/src/main/java/org/mozilla/fenix/components/metrics/AdjustMetricsService.kt

```diff
@@ -88,9 +88,13 @@ class AdjustMetricsService(
     override fun track(event: Event) {
         CoroutineScope(dispatcher).launch {
             try {
-                if (event is Event.GrowthData && storage.shouldTrack(event)) {
-                    Adjust.trackEvent(AdjustEvent(event.tokenName))
-                    storage.updateSentState(event)
+                if (event is Event.GrowthData) {
+                    if (storage.shouldTrack(event)) {
+                        Adjust.trackEvent(AdjustEvent(event.tokenName))
+                        storage.updateSentState(event)
+                    } else {
+                        storage.updatePersistentState(event)
+                    }
```

Comment:
Yeah, I'm loving how clear this is now 🙌 thanks for updating!

Reply:
yeah this new flow is much better and easier to expand on later if need be. thank you for your advice!